commons/src/main/java/com/powsybl/commons/datasource/TarArchiveDataSource.java

```diff
@@ -0,0 +1,329 @@
+            return new TarArchiveOutputStream(new BufferedOutputStream(Files.newOutputStream(tmpTarFilePath)));
+        }
+
+        private static Path getTmpTarFilePath(Path tarFilePath) {
+            return tarFilePath.getParent().resolve("tmp_" + tarFilePath.getFileName());
+        }
+
+        private static Path getTmpCompressedTarFilePath(Path tarFilePath) {
+            return tarFilePath.getParent().resolve("tmp_comp_" + tarFilePath.getFileName());
+        }
+
+        private void compressTarFile() throws IOException {
+            try (InputStream fis = Files.newInputStream(getTmpTarFilePath(tarFilePath));
+                 OutputStream fos = Files.newOutputStream(getTmpCompressedTarFilePath(tarFilePath), StandardOpenOption.CREATE);
+                 OutputStream compressedOS = getCompressedOutputStream(fos, this.compressionFormat)) {
+                byte[] buffer = new byte[8192];
+                int len;
+                while ((len = fis.read(buffer)) != -1) {
+                    compressedOS.write(buffer, 0, len);
+                }
+            }
+        }
+
+        private static OutputStream getCompressedOutputStream(OutputStream os, CompressionFormat compressionFormat) throws IOException {
+            return compressionFormat == null ? os : switch (compressionFormat) {
+                case GZIP -> new GzipCompressorOutputStream(os);
+                case BZIP2 -> new BZip2CompressorOutputStream(os);
+                case XZ -> new XZCompressorOutputStream(os);
+                case ZSTD -> new ZstdCompressorOutputStream(os);
+                default -> os;
+            };
+        }
+
+        @Override
+        public void close() throws IOException {
+            if (!closed) {
+
+                // Close temporary stream file
+                super.close();
+
+                // Open a new temporary archive
+                try (TarArchiveOutputStream taos = getTarStream(getTmpTarFilePath(tarFilePath))) {
+
+                    // Useful parameter
+                    taos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_POSIX);
+                    taos.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
+
+                    // Temporary stream file path
+                    Path tmpStreamFilePath = getTmpStreamFilePath(tarFilePath);
+
+                    // Copy content of temporary stream file into an entry of the temporary archive
+                    try (InputStream is = Files.newInputStream(tmpStreamFilePath)) {
+
+                        // New tar entry
+                        TarArchiveEntry entry = new TarArchiveEntry(fileName);
+                        entry.setSize(Files.size(tmpStreamFilePath));
+
+                        // New file to add
+                        taos.putArchiveEntry(entry);
+
+                        // Write the data in the entry
+                        taos.write(is.readAllBytes());
```

Comment:
The problem remains the same here. We could call the same method as in `ZipArchiveDataSource`
```suggestion
                        ByteStreams.copy(is, taos);
```